File: core/evidence/worldview_variance_engine.py
from __future__ import annotations

from typing import Any, Dict, List


def model_worldview_variance(interpretation_count: int, contradiction_pairs: int) -> Dict[str, Any]:
    variance = round(min(1.0, interpretation_count * 0.2 + contradiction_pairs * 0.15), 3)
    return {"variance": variance, "preserved": variance > 0, "convergence_blocked": True}

File: src/evidence/worldviewVarianceEngine.ts
/**
 * Converted from Python: core/evidence/worldview_variance_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function modelWorldviewVariance(interpretation_count: any, contradiction_pairs: any): any {
  var variance: any = py.round(py.min([py.F(1.0), py.add(py.mul(interpretation_count, py.F(0.2)), py.mul(contradiction_pairs, py.F(0.15)))]), 3);
  return {"variance": variance, "preserved": (variance > 0), "convergence_blocked": true};
}
